### DISCUSSION TITLE
Scene inheritance needs usability improvements

### DISCUSSION THREAD

**Discussion by peti29** [godotengine/godot-proposals#7975] [Editor]
I'm very new to Godot, please bear with me.

I'm also not well versed in game development. I am however an experienced software engineer.
I only say this to let you know that I am familiar with the concept of inheritance. (In fact I already use it in gd scripts.)

But scene inheritance just doesn't sit well with me.

- It should be more discoverable. It took me some time to notice that the parent of an inherited scene is visible after all (in a tooltip if you hover over it with the mouse). I think this information should be more pronounced.
Also, inherited scenes should be more clearly indicated. ("Some nodes are sometimes yellow" - is not enough IMO)
I just discovered now that there is a way after all to navigate to the parent node, but it's called "Open in Editor" in the child scene's context menu. It should be called "Open Parent in Editor" instead.

- It's too much behind the scenes (no pun intended!). Admittedly I somehow messed up my project so that I was unable to modify keys in an AnimationPlayer of the child scene, but that AnimationPayer was not even present in the parent node. Godot still insisted that it was an inherited node and wouldn't let me edit it. I wasn't able to discover what went wrong and needed to recreate my scenes (this time carefully avoiding scene inheritance!)

- There could be a tree diagram displaying scene hierarchy (though that may be an overkill).

- It would be nice if individual nodes could be toggled between "inherit the relevant node from parent scene" and "override relevant node in the child scene"

- It would be nice to be able  to "reroute" scenes: specify a new / different / no parent

**Comment by hunterloftis** [comment]
My primary interest here is:

> It would be nice to be able to "reroute" scenes: specify a new / different / no parent

This is currently a huge pain that involves making duplicates of scenes & nodes, then copy pasting from the old scene to the new scene. An accessible UI for this (like right-click scene -> change parent) is needed.
